internal/client/errors.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type apiResponseError struct {
+	body   string
+	status uint16
+}
+
+func newAPIError(resp *http.Response) apiResponseError {
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+
+	status := uint16(resp.StatusCode)
+	body := ""
+	if err != nil {
+		body = fmt.Sprintf("failed to read response body. reason: %s", err.Error())
+	} else {
+		body = string(respBody)
+	}
+
+	return apiResponseError{
+		body:   body,
+		status: uint16(status),
```

Comment:
nit: The type was already cast to `uint16` on line 19. No need to cast it here or we can remove line 19.